### PR TITLE
Feat/detect root files

### PR DESCRIPTION
**Pull request by erfemega**
- Adding detectRootFiles API
- Adding related tests

### CODE REVIEW

File: lib/detectRoot.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        type: 'WSDL',
+        version: version
+      },
+      data: adaptedData
+    }
+  };
+}
+
+module.exports = {
+  getRootFiles: (inputData, wsdlParser, xmlParser) => {
```
Comment (LuisTejedaS) [inline lib/detectRoot.js:30]: We can add documentation to this method

File: lib/utils/multifile.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        },
+        data: adaptedData
+      }
+    };
+  },
+
+  /**
+   * Maps the input from detect root files to get root files
+   * @param {object} input - input schema
+   * @returns {Array} - Array of all MISSING_ENDPOINT objects
```
Comment (LuisTejedaS) [inline lib/utils/multifile.js:99]: this description is incorrect

File: test/unit/detectRootFiles.test.js [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+        ]
+      },
+      result = await Converter.detectRootFiles(input);
+    expect(result.result).to.be.true;
+    expect(result.output.data[0].path).to.equal('/CountingCategoryService.wsdl');
+    expect(result.output.specification.version).to.equal('1.1');
+    expect(result.output.type).to.be.equal('rootFiles');
+  });
+
+  it('should return no root when specific version is not present', async function() {
```
Comment (LuisTejedaS) [inline test/unit/detectRootFiles.test.js:78]: should be:

should return no root when a specific version is 2.0 and no root of that version is present

**Review by LuisTejedaS — COMMENTED** [review]
just a couple of comments

**Review by LuisTejedaS — APPROVED** [review]
LGTM

**Review by Carlos-Veloz — APPROVED** [review]

**Pull request merged**